src/Plugin/Field/FieldWidget/ExploreCoursesUrlWidget.php

```diff
@@ -0,0 +1,188 @@
+      $container->get('http_client')
+    );
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public static function defaultSettings() {
+    $settings = [
+      'api_version' => '20200810',
+    ];
+    return $settings + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, ClientInterface $client) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->client = $client;
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements['api_version'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('API version'),
+      '#required' => TRUE,
+      '#default_value' => $this->getSetting('api_version'),
+      '#access' => TRUE,
+      '#element_validate' => [
+        [$this, 'validateApi'],
+      ],
+    ];
+    return $elements;
+  }
+
+  /**
+   * Validate we are specifying the newest api.
+   *
+   * @param array $element
+   *   Url form element.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Current form state object.
+   * @param array $complete_form
+   *   Complete form.
+   */
+  public function validateApi(array &$element, FormStateInterface $form_state, array &$complete_form) {
+    $input = NestedArray::getValue($form_state->getValues(), $element['#parents']);
+
+    if ($form_state::hasAnyErrors()) {
+      return;
+    }
+    try {
+      $response = $this->client->request('GET', 'search?view=xml-' . $input, ['base_uri' => 'https://explorecourses.stanford.edu/']);
+      $response = (string) $response->getBody();
+      $xml = new \SimpleXMLElement($response);
+      // Do this as a string, since SimpleXMLElement doesn't cast to bools.
+      if ((string) $xml->deprecated == 'true') {
+        $form_state->setError($element, $this->t("That API version is deprecated. Newest version is: $xml->latestVersion"));
+      }
+    }
+    catch (\Throwable $e) {
+      $form_state->setError($element, $this->t('There was a problem querying the ExploreCourses API.'));
+    }
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public function settingsSummary() {
+    $summary = [];
+    if (empty($this->getSetting('api_version'))) {
+      $summary[] = $this->t('No API version Provided');
+    }
+    else {
+      $summary[] = $this->t('API version: @api_version', ['@api_version' => $this->getSetting('api_version')]);
+    }
+    return $summary;
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+
+    $element['#element_validate'] = [
+      [$this, 'validateExploreCourseUrl'],
+    ];
```

Comment:
This way it doesn't remove any existing validators. Are you sure you don't want to set this on the `uri` part to be more specific?
```suggestion
    $element['#element_validate'][] = [$this, 'validateExploreCourseUrl'];
```

Reply:
fixed.